### ISSUE TITLE
iterator_to_array($iterable)[0] misuse false positive when iterable is not Iterator

### ISSUE THREAD

**Issue by someniatko** [kalessil/phpinspectionsea#1302]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate) v2.0.15 |
| Language level | PHP 7.2 |

Current behaviour
--
Consider the following snippet:

```php
<?php

class Collection implements IteratorAggregate {
    public function getIterator()
    {
        yield from [ 1, 2, 3 ];
    }
}

$collection = new Collection;
iterator_to_array($collection)[0];
```

At the last line, an inspection "Consider using '$collection->current()' instead" suggestion shows up. However, the $collection instance does not have method "current()" therefore the suggestion leads to invalid code.

Expected behaviour
--
The mentioned inspection is only fired if $iterable is guaranteed to be instance of Iterator.

Environment details
---
PhpStorm 2019.1.1
Build #PS-191.6707.66, built on April 17, 2019
Licensed to <me>
Subscription is active until <date>
JRE: 1.8.0_202-release-1483-b44 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

**Comment by kalessil** [comment]
Fixed!
